test/e2e/network/netpol/k8s_util.go

```diff
@@ -0,0 +1,352 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package netpol
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// Scenario provides a convenience interface to kube functionality that we leverage for polling NetworkPolicy connections.
+type Scenario struct {
+	mutex     *sync.Mutex
+	podCache  map[string][]v1.Pod
+	framework *framework.Framework
+	ClientSet clientset.Interface
+}
+
+// NewScenario is a utility function that wraps creation of the stuff we're using for creating a cluster that expresses the global policy scenario we're testing.
+func NewScenario(framework *framework.Framework, clientSet clientset.Interface) *Scenario {
```

Comment:
one options is to remove the clientset argument of the constructor, but keep it in the struct and get it from the `framework` variable

Reply:
ic, you are already passing the framework clientset when calling it ... so up to you then if you want to simplify it

Reply:
👍